### PR TITLE
Adding Zoom and ColorSub_C/M/Y GDTF attributes, Pigtail settings, catch several crash errors

### PR DESCRIPTION
**Pull request by vanous**
This adds several things:

* initial Zoom support
* initial ColorSub_C/M/Y support
* fixed the ColorAdd_R/G/B attribute support (not sure how this worked before... perhaps the original devices used R, G, B instead of the GDTF attributes...?)
* added Pigtail settings - this is not updated in realtime but only during adding/editing of a device
* catch several crash errors

Please let me know if i miss some of the formats or variable camelCasing.

### CODE REVIEW

**Review by hugoaboud — CHANGES_REQUESTED** [review]
Great job! I have a few comments and questions, but overall it looks very good.

File: gdtf.py [hunk 1]
```diff
@@ -157,10 +168,13 @@ def load3ds(profile, model):
         obj = bpy.context.view_layer.objects.selected[0]
         obj.users_collection[0].objects.unlink(obj)
         obj.rotation_euler = Euler((0, 0, 0), 'XYZ')
+        z=obj.dimensions.z
```
[the remark this answers is not in the document]
Comment (vanous) [reply gdtf.py:171]: uh, i forgot python has this shorthand :)

File: gdtf.py [hunk 2]
```diff
@@ -210,9 +221,13 @@ def updateGeom(geom, d=0):
                 obj_child.location[1] += position[1]
                 obj_child.location[2] += position[2]
                 scale = [geom.position.matrix[c][c] for c in range(3)]
-                obj_child.scale[0] *= scale[0]
                 obj_child.scale[1] *= scale[1]
                 obj_child.scale[2] *= scale[2]
+                if 'Pigtail' in geom.model:
```
[the remark this answers is not in the document]
Comment (vanous) [reply gdtf.py:226]: There can be multiple Pigtails easily. Larger linear devices indicate where input/outputs are (if on each side of the device) other devices might be plug-able in the base and then also in the yoke (screen carriers).

File: gdtf.py [hunk 3]
```diff
@@ -210,9 +221,13 @@ def updateGeom(geom, d=0):
                 obj_child.location[1] += position[1]
                 obj_child.location[2] += position[2]
                 scale = [geom.position.matrix[c][c] for c in range(3)]
-                obj_child.scale[0] *= scale[0]
```
[the remark this answers is not in the document]
Comment (vanous) [reply gdtf.py:213]: Wow, my bad, thank you for catching this. Apologies.

**Comment by vanous** [comment]
OK, this should be ready now.

**Review by hugoaboud — APPROVED** [review]

**Comment by hugoaboud** [comment]
LGTM

**Pull request merged**